### PR TITLE
Fixed z-fighting issue and integrated mouse input with camera

### PR DESCRIPTION
**Pull request by ttanatb**
## Feature/Issue
Fixed z-fighting issue and integrated mouse input with camera

## Implementation/Solution
Fixed z-fighting issue and integrated mouse input with camera

## Dependencies (if any?)
Fixed z-fighting issue and integrated mouse input with camera
 
## Screenshots/GIF (if applicable)
Fixed z-fighting issue and integrated mouse input with camera

## Tests
Fixed z-fighting issue and integrated mouse input with camera

### CODE REVIEW

File: Light Vox Engine/Engine.cpp [hunk 1]
```diff
@@ -14,16 +14,17 @@
...
 {
-	if (hInstance != nullptr) {
-		this->hInstance = hInstance;
-		Engine::engineInstance = this;
-		windowWidth = 1280;
-		windowHeight = 720;
-		windowTitle = "STRUGGLE BUS";
-		hWindow = 0;
-	}
+    if ( hInstance != nullptr )
```
Comment (BenjaFriend) [inline Light Vox Engine/Engine.cpp:19]: hInstance is Windows only

File: Light Vox Engine/Engine.cpp [hunk 2]
```diff
@@ -133,10 +134,8 @@ LV_RESULT Engine::InitSystems()
     }
 
     //DEBUG:: INTIALIZE ENTITY POSSITIONS
-    static int count = static_cast<int>( sqrtf( static_cast<float>( LV_MAX_INSTANCE_COUNT ) ) );
-    static float rotation = 0.001f;
     {
-
+        static int count = static_cast<int>( sqrtf( static_cast<float>( LV_MAX_INSTANCE_COUNT ) ) );
```
Comment (BenjaFriend) [inline Light Vox Engine/Engine.cpp:138]: `sqrt` is gross but this is an init function so it should be fine. All this casting could produce weird results with a non-square number though.

File: Light Vox Engine/Graphics/Camera.h [hunk 3]
```diff
@@ -9,17 +12,33 @@ namespace Graphics
...
+
         float pitchAngle;
+        float yawAngle;
+
         float fov;      //field of view
         float nearZ;    //near plane (in Z)
         float farZ;     //far plane (in Z)
 
-        const float MAX_PITCH = glm::pi<float>() / 2.0f;
+        Input::InputManager *inputManager;
```
Comment (BenjaFriend) [inline Light Vox Engine/Graphics/Camera.h]: I like to explicitly initialize things to `nullptr`

**Review by BenjaFriend — DISMISSED** [review]
Fine overall, just some semantic suggestions.

File: Light Vox Engine/Engine.cpp (hunk 1 above)
Comment (ttanatb) [reply Light Vox Engine/Engine.cpp:19]: it's an LV_INSTANCE so it'll be typedef'ed to the appropriate thing on Windows and differently on other platforms

**Review by BenjaFriend — APPROVED** [review]
🚢

**Pull request merged**